core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -37,6 +37,17 @@ class GameInfo {
     var currentTurnStartTime = 0L
     var gameId = UUID.randomUUID().toString() // random string
 
+    fun addTechnologyToTeam(team: String, techName: String) {
+        var teammatesCivs = civilizations - civilizations.filter { it.team != team }
+        for (teammate in teammatesCivs) {
+            if (teammate.tech.techsResearched.find { it == techName } == techName)
+                teammatesCivs -= teammate
```

Comment:
maybe I should remove the teammate from Civs after the for? Would this risk a ConcurrentModificationException?